streams/src/main/java/org/apache/kafka/streams/processor/PartitionGrouper.java

```diff
@@ -26,19 +26,71 @@
 
 public abstract class PartitionGrouper {
 
-    protected Map<Integer, Set<String>> topicGroups;
+    public static class TopicsInfo {
+        public Set<String> sourceTopics;
+        public Set<String> stateTopics;
+
+        public TopicsInfo(Set<String> sourceTopics, Set<String> stateTopics) {
+            this.sourceTopics = sourceTopics;
+            this.stateTopics = stateTopics;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (o instanceof TopicsInfo) {
+                TopicsInfo other = (TopicsInfo) o;
+                return other.sourceTopics.equals(this.sourceTopics) && other.stateTopics.equals(this.stateTopics);
+            } else {
+                return false;
+            }
+        }
+
+        @Override
+        public int hashCode() {
+            long n = ((long) sourceTopics.hashCode() << 32) | (long) stateTopics.hashCode();
+            return (int) (n % 0xFFFFFFFFL);
+        }
+    }
+
+    public static class TasksInfo {
+        public Map<TaskId, Set<TopicPartition>> partitionsForTask;
+        public Map<String, Set<TaskId>> tasksForState;
```

Comment:
I don't like that PartitionGrouper being concerned with state store change log topics. It seems possible to move the logic in DefaultPartitionGrouper to KafkaStreamingPartitionAssigner and to make PartitionGrouper only do grouping of partitions. PartitionGrouper something that a user can implement. It doesn't seem right to have the code related to change log partition creation there. It is not meant to be customized, right?